frontend/src/concepts/pipelines/content/createRun/contentSections/MaxConcurrencyField.tsx

```diff
@@ -21,7 +21,7 @@ const MaxConcurrencyField: React.FC<MaxConcurrencyFieldProps> = ({ onChange, val
       data-testid="maxConcurrencyField"
       min={1}
       max={10}
-      onChange={onChange}
+      onChange={(newValue) => onChange(newValue ?? 0)}
```

Comment:
Should we put this on everyone or give `NumberInput` a prop that can provide the fallback value and have it default to `0` as previously done?

Reply:
We need it to come back as `undefined` for inputs that want to be able to unset the value after it has been set (undefined rather that 0). Allowing the fallback to be `undefined` would require this change anyhow (though we'd know it would never really receive undefined).